src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+
+/**
+ * The Parser class is in charge of handling user-given commands. It ensures
+ * that only valid inputs are processed and produces the relevant Command
+ * objects.
+ */
+public class Parser {
+
+    /**
+     * Parses the input string and returns a Command corresponding to the
+     * required action.
+     *
+     * @param input             String representation of the desired command
+     * @return                  Command
+     * @throws DukeException    If input is invalid, etc.
+     */
+    public static Command parse(String input) throws DukeException {
+        String command = input.strip();
+        validate(command);
+
+        if (command.startsWith("done")) {
+            return new DoneCommand(command.substring(5), true);
+        } else if (command.startsWith("undo")) {
+            return new DoneCommand(command.substring(5), false);
+        } else if (command.startsWith("delete")) {
+            return new DeleteCommand(command.substring(7));
+        } else if (command.startsWith("todo")) {
+            return new TodoCommand(command.substring(5));
+        } else if (command.startsWith("event")) {
+            return new EventCommand(command.substring(6));
+        } else if (command.startsWith("deadline")) {
+            return new DeadlineCommand(command.substring(9));
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.startsWith("find")) {
+            return new FindCommand(command.substring(5));
+        } else { // input is "bye"
+            return new ByeCommand();
+        }
+    }
+
+    /**
+     * Throws an exception if the given input does not have a valid format.
```

Comment:
This could be the description for @throws DukeException part. A better way to describe this method would be something like "Validates a command" or "Checks for valid format in a command" etc.